text/0000-cargo-prepublish.md

```diff
@@ -0,0 +1,395 @@
+- Feature Name: prepublish
+- Start Date: 2017-03-22
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes the concept of *augmenting sources* for Cargo. Sources can be
+enhanced with new versions of a crate that did not exist or have existing
```

Comment:
My brain is having trouble parsing this sentence, and I'm not sure what it's saying:

> Sources can be enhanced with new versions of a crate that did not exist or have existing versions replaced.

I think this is supposed to expand out to mean:

> Sources can be enhanced with new versions of a crate that did not exist, or sources can have existing versions replaced.

Or perhaps:

> Sources can be enhanced by adding new versions of a crate that do not exist or by replacing existing versions

The less likely but still possible reading is something like:

> Sources can be enhanced with new versions of a crate that did not exist
> Sources can be enhanced with new versions of a crate that have existing versions replaced

due to the ambiguous subject of the second clause and the verbs that don't quite match...